### ISSUE TITLE
debian install refuses to log to remote influxdb

### ISSUE THREAD

**Issue by claytondukes** [influxdata/telegraf#1631]
## Bug report
Perhaps I am losing my mind, but telegraf installed on debian using the same config as I have on another Ubuntu server will not send metrics to a remote influxdb server.
The log shows:
```
016/08/13 17:32:30 Database creation failed: Get http://localhost:8086/query?db=&q=CREATE+DATABASE+%22telegraf%22: dial tcp 127.0.0.1:8086: getsockopt: connection refused
2016/08/13 17:39:50 Error writing to output [influxdb]: Could not write to any InfluxDB server in cluster
```

### Relevant telegraf.conf:
```
[global_tags]
[agent]
  interval = "10s"
  round_interval = true
  metric_batch_size = 1000
  metric_buffer_limit = 10000
  collection_jitter = "0s"
  flush_interval = "10s"
  flush_jitter = "0s"
  debug = true
  quiet = false
  hostname = ""
  omit_hostname = false
[[outputs.influxdb]]
  urls = ["http://192.168.28.118:8086"] # required
  database = "telegraf" # required
  precision = "s"
  retention_policy = "default"
  write_consistency = "any"
  timeout = "5s"
[[inputs.cpu]]
  percpu = true
  totalcpu = true
  fielddrop = ["time_*"]
[[inputs.disk]]
  ignore_fs = ["tmpfs", "devtmpfs"]
[[inputs.diskio]]
[[inputs.kernel]]
[[inputs.mem]]
[[inputs.processes]]
[[inputs.swap]]
[[inputs.system]]
[[inputs.dns_query]]
   servers = ["8.8.8.8"] # required
   domains = ["."] # optional
   #response_timeout = "5s"
   #method = "GET"
[[inputs.net]]
[[inputs.ntpq]]
   dns_lookup = true
 [[inputs.ping]]
   urls = ["www.google.com"] # required
   count = 1 # required
   ping_interval = 0.0
   timeout = 2.0
   interface = "vmbr0"
```
### System info:
Telegraf - version 0.13.2
Debian Jessie

### Steps to reproduce:

```
service telegraf start
tail -f /var/log/telegraf/telegraf.log
```

### Expected behavior:
Should send data to the remove influxdb server at http://192.168.28.118:8086

### Actual behavior:
Seems to ignore that setting for some reason and insists on trying to send to localhost

### Additional info:

Here's the strange part. If I start using the exact same command found in /lib/systemd/system/telegraf.service, it runs fine:
```
root@bfs [telegraf]: # /bin/sh -c "/usr/bin/telegraf -config /etc/telegraf/telegraf.conf"
2016/08/13 17:47:16 Attempting connection to output: influxdb
2016/08/13 17:47:16 Successfully connected to output: influxdb
2016/08/13 17:47:16 Starting Telegraf (version 0.13.2)
2016/08/13 17:47:16 Loaded outputs: influxdb
2016/08/13 17:47:16 Loaded inputs: processes ping disk diskio kernel mem swap system dns_query net cpu ntpq
2016/08/13 17:47:16 Tags enabled: host=bfs
2016/08/13 17:47:16 Agent Config: Interval:10s, Debug:true, Quiet:false, Hostname:"bfs", Flush Interval:10s
2016/08/13 17:47:20 Input [mem] gathered metrics, (10s interval) in 1.160859ms
2016/08/13 17:47:20 Input [swap] gathered metrics, (10s interval) in 522.762µs
2016/08/13 17:47:20 Input [kernel] gathered metrics, (10s interval) in 11.614062ms
2016/08/13 17:47:20 Input [disk] gathered metrics, (10s interval) in 18.15228ms

```

I have searched and the system has no other telegraf.conf files. And if I move the /etc/telegraf/telegraf.conf file, service telegraf start complains and won't start, so I know that it is at least reading that file....partially?

**Comment by sparrc** [comment]
I can't fathom how this would be possible, here are the precedents for config files, did you check all of these?:

```
  1. -config flag
  2. $TELEGRAF_CONFIG_PATH environment variable
  3. $HOME/.telegraf/telegraf.conf
  4. /etc/telegraf/telegraf.conf
```

also check /etc/telegraf/telegraf.d

**Comment by claytondukes** [comment]
Nor could I :)
So here's what I found this morning:
In Debian, the `service telegraf stop` function is not working, thus it was not actually re-reading the new config.
After killing the pids:
`for pid in $(pidof /usr/bin/telegraf); do kill -9 $pid; done` 
and running `service telegraf start`, it works.

**Comment by sparrc** [comment]
okay, there is an issue with systemd, it is fixed in the 1.0 beta releases
